js/anims-missiles.js

```diff
@@ -0,0 +1,112 @@
+repeatAnims()
+
+function repeatAnims() {
+    setInterval(function() {
+        var animationTime = animationChangeSpeed()
+        dropMissile(animationTime)
+    }, 1000)
+}
+
+function dropMissile(animationTime) {
+    var columnNumber = randomNumGen(4)
+    var $missile = $('<img src="img/missile-drop-down-white.png" ' +
+        'class="missile missile-'+ columnNumber +'">')
+    $('#pos-'+columnNumber).prepend($missile)
+
+    $missile.animate({
+        top: "+=550"
+    }, animationTime, "linear", function () {
+        missileHitsCity()
+        $(this).remove()
+    })
+}
+
+function missileHitsCity() {
+    gameScore.score -= 1
+    gameScore.toll += 10000
+    document.querySelector('#score').textContent = gameScore.score
+    document.querySelector('#toll').textContent = gameScore.toll
+}
+
+function randomNumGen(topLimit) {
+    var randomNum = Math.ceil(Math.random() * topLimit)
+    return(randomNum)
+}
+
+function createMissileEventHandler() {
+    var cities = document.querySelectorAll('.city')
+
+    cities.forEach(function(city) {
+        city.addEventListener('click', function() {
+            columnID = this.getAttribute('data-city')
+            isHit()
+        })
+    })//end forEach
+
+}
+
+function isHit() {
+    var cityNumber = columnID
+    var missileNumber = '.missile-' + cityNumber
+    var missiles = document.querySelectorAll(missileNumber)
+    var hasMissiles = document.querySelectorAll(missileNumber).length
+    if (hasMissiles) {
+        gameScore.score += hasMissiles
+        document.querySelector('#score').textContent = gameScore.score
+        $(missileNumber).stop()
+        missiles.forEach(function (missile) {
+            missile.classList.remove(missileNumber)
+            missile.src = "img/missile-explosion.gif";
+            setTimeout(function () {
+                try {
+                    missile.parentNode.removeChild(missile);
+                } catch (e) {
+                    // squash errors where the missile is already exploding and we don't need to remove it.
+                }
+            }, 500)
+        })
+    } else {
+        gameScore.score -= 1
+        document.querySelector('#score').textContent = gameScore.score
+    }
+}
+
+function listenKeypressMissiles() {
+    clicked = false
+    document.addEventListener('keypress', function(e) {
+        var keys = {
+            'q': '1',
+            'w': '2',
+            'e': '3',
+            'r': '4'
+        }
+        if (e.key in keys) {
+            columnID = keys[e.key]
+            isHit()
+        }
+    })
+}
+
+// Increases animation speed based upon gameScore.score
+// We call this function inside repeatAnim function
+function animationChangeSpeed() {
+    var animationDuration = 3000
+    if (gameScore.score > 0) {
+        animationDuration = 3000 / (1 + (gameScore.score * 0.03))
+    } else {
+        animationDuration = 3000
+    }
+    return animationDuration
+}
+
+createMissileEventHandler()
```

Comment:
Perhaps group functions and have functions called all in the same place?

Reply:
Agreed - this will be done when we implement the start button.